api/modules/trips/views.py

```diff
@@ -92,7 +92,13 @@ def add_friend_to_trip(request, trip_id, user_id):
             error_message = "User already associated with trip"
             return Response(error_message, status=status.HTTP_400_BAD_REQUEST)
 
-        trip.users.add(user)
+        notification_text = "You are added in a trip."
+        if add_trip_notification(request.user, user, trip, notification_text):
+            trip.users.add(user)
```

Comment:
If notification succeeds but trip fails, you will have a false notification.
Add a trip first, then notification so that even if user doesn't get notified still he should have the trip information